hedera-node/hedera-app/src/main/java/com/hedera/node/app/records/impl/BlockRecordManagerImpl.java

```diff
@@ -370,11 +371,11 @@ public void advanceConsensusClock(@NonNull final Instant consensusTime, @NonNull
         final var newBlockInfo = builder.build();
 
         // Update the latest block info in state
-        final var states = state.getWritableStates(BlockRecordService.NAME);
+        final var states = stack.getWritableStates(BlockRecordService.NAME);
         final var blockInfoState = states.<BlockInfo>getSingleton(V0490BlockRecordSchema.BLOCK_INFO_STATE_KEY);
         blockInfoState.put(newBlockInfo);
         // Commit the changes. We don't ever want to roll back when advancing the consensus clock
-        ((WritableSingletonStateBase<BlockInfo>) blockInfoState).commit();
+        stack.commitFullStack();
```

Comment:
Do we really want to commit the full stack every time we advance the consensus clock? The removed line is only scoped to the block info state, whereas the savepoint stack could potentially include all of state

Reply:
Yes, we should always view all parts of the state and their changes as a whole to ensure the committed data is consistent. If we ever come to a situation where committing only part of the state is necessary, something is awfully wrong.

Reply:
good to know, thanks for the context!